File: ruby/word-count/word_count.rb
class Phrase
  def initialize(words)
    @array_of_words = words.downcase.split(/[^'\w]+/)
  end

  def word_count
    @array_of_words.uniq.to_h { |word| [word, @array_of_words.count(word)] }
  end
end

File: elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
    clean_sentence = String.split(sentence, ["_",",","!","&","@","$","%","^","&",":", " "], trim: true)
    Enum.frequencies_by(clean_sentence, &String.downcase/1)
  end
end
